uplink/decorators.py

```diff
@@ -103,10 +103,17 @@ class headers(MethodAnnotation):
     """
 
     def __init__(self, arg, **kwargs):
-        # TODO: allow the first argument to be a list or str, in
-        # which case you would split the strings by a colon delimiter
-        # to identify the headers.
-        self._headers = dict(arg, **kwargs)
+        if isinstance(arg, str):
+            key, value = map(str.strip, arg.split(":"))
```

Comment:
I mention this again below: you should move `map(str.stip, arg.split(":")` into a separate method (e.g., `_get_header`), so that you can also use it in the `elif` condition.
```python
key, value = self._get_header(arg)
```